### DISCUSSION TITLE
Installing azure-cli

### DISCUSSION THREAD

**Discussion by jeet-proj** [Homebrew/discussions#1485] [Getting started]
Hi,
I am trying to install azure-cli on my mac that is running macOS: 10.15.7-x86_64
My installation of brew runs successfully and when I try to update, it does say that its up to date
When I try searching azure-cli, it fails to find.
If I do 'brew search azure' it just returns the values from 'Casks' and not from 'Formulae' 
Where as when my colleague tries, he does get results back from 'Formulae' where 'azure-cli' is available.
It will be a great help if someone can figure out the cause of it. Or suggest me to install azure-cli differently.

Regards,

Jeet

**Comment by carlocab** [comment]
What's the output of `brew config` and `brew doctor`?

**Reply by jeet-proj** [reply]
Sorry for coming back a bit late 
brew config 
HOMEBREW_VERSION: 3.1.7-84-ge0bfcda
ORIGIN: https://github.com/Homebrew/brew
HEAD: e0bfcdaf258dadaaf8c102d4cb51f23b6fdb2c4c
Last commit: 80 minutes ago
Core tap ORIGIN: https://github.com/Homebrew/brew
Core tap HEAD: e0bfcdaf258dadaaf8c102d4cb51f23b6fdb2c4c
Core tap last commit: 80 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
CPU: quad-core 64-bit ivybridge
Clang: 12.0.0 build 1200
Git: 2.24.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 10.15.7-x86_64
CLT: 12.0.0.32.29
Xcode: 12.4


brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Suspicious https://github.com/Homebrew/homebrew-core git origin remote found.
The current git origin is:
  https://github.com/Homebrew/brew

With a non-standard origin, Homebrew won't update properly.
You can solve this by setting the origin remote:
  git -C "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core" remote set-url origin https://github.com/Homebrew/homebrew-core

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /Library/Frameworks/Python.framework/Versions/3.9/bin/python3.9-config
  /Library/Frameworks/Python.framework/Versions/3.9/bin/python3-config
  /Library/Frameworks/Python.framework/Versions/3.6/bin/python3.6m-config
  /Library/Frameworks/Python.framework/Versions/3.6/bin/python3-config
  /Library/Frameworks/Python.framework/Versions/3.6/bin/python3.6-config

**Reply by jeet-proj** [reply]
After removing the Pyhton 
brew doctor returns this

---
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Suspicious https://github.com/Homebrew/homebrew-core git origin remote found.
The current git origin is:
  https://github.com/Homebrew/brew

With a non-standard origin, Homebrew won't update properly.
You can solve this by setting the origin remote:
  git -C "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core" remote set-url origin https://github.com/Homebrew/homebrew-core

**Reply by carlocab** [reply]
Have you tried the `git` command suggested by `brew doctor`? What happens when you do?

**Reply by jeet-proj** [reply]
yes, it doesn't fix it. When run again. I see the same issue

**Reply by jeet-proj** [reply]
sorry the suggestion changes to this 
git -C "/usr/local/Homebrew" remote set-url origin https://github.com/Homebrew/brew

**Reply by jeet-proj** [reply]
If I then do that.. it goes back to the previous suggestion

**Reply by jeet-proj** [reply]
my search result just with this
$ brew search azure
==> Casks
azure-data-studio                                                       microsoft-azure-storage-explorer                                        homebrew/cask-versions/azure-data-studio-insiders

**Reply by carlocab** [reply]
Your core tap is broken. Do
```
git -C "/usr/local/Homebrew" remote set-url origin https://github.com/Homebrew/brew
rm -rf $(brew --repository homebrew/core)
brew tap homebrew/core
```

**Reply by jeet-proj** [reply]
brew tap fails, tried twice. Broadband speed at my end is 100mbps 
==> Tapping homebrew/core
Cloning into '/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core'...
remote: Enumerating objects: 959369, done.
remote: Counting objects: 100% (204/204), done.
remote: Compressing objects: 100% (95/95), done.
fatal: the remote end hung up unexpectedly59.32 MiB | 10.88 MiB/s
fatal: early EOF
fatal: index-pack failed
Error: Failure while executing; `git clone https://github.com/Homebrew/homebrew-core /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core --origin=origin --template=` exited with 128.

**Reply by jeet-proj** [reply]
tried multiple times

**Reply by jeet-proj** [reply]
Fails in cloning at around 74%. Will try again in evening. Thanks for helping. If there are any other things that I could do, please let me know and I will give it a go

**Reply by carlocab** [reply]
Try
```
cd /tmp
git clone https://github.com/Homebrew/homebrew-core 
```
if that fails, then it might be a connection problem. See #666 for tips on what other users have tried to resolve those. If it works, try `brew tap homebrew/core` again. If that still doesn't work (but cloning into `/tmp` succeeded), try
```
cd /usr/local/Homebrew/Library/Taps/homebrew
git clone https://github.com/Homebrew/homebrew-core
```

**Reply by jeet-proj** [reply]
I already had increased the postbuffer size as per this [link](https://stackoverflow.com/questions/6842687/the-remote-end-hung-up-unexpectedly-while-git-cloning) but still fails.. its around 74-78% when it fails

**Reply by jeet-proj** [reply]
The solution which is a bit odd as it's a workaround than a pure solution
On Mac "git clone https://github.com/Homebrew/homebrew-core" failed and I tried every help that was available but cloning failed. I tried more than 30 times over 2 days. It never succeeded more than 74%.

On windows, the same command worked first time and every time. Luckily I had a windows and I hate to say windows worked better than Mac this time

It created 'homebrew-core' folder which I copied across to Mac on /Users/<my name> and then moved the folder across to "/usr/local/Homebrew/Library/Taps/homebrew" 
Make sure to move the folder "homebrew-core" inside homebrew and not just the files.
Thus now I had 2 folders inside 
/usr/local/Homebrew/Library/Taps/homebrew
     homebrew-cask
     homebrew-core
Ran 'brew install azure-cli' and it updated brew as well as successfully installed azure-cli

**Reply by SMillerDev** [reply]
What does `brew config` and `brew doctor` say now? Because there is a chance that did not move the repository information and in that case you'll have to do this more often.

**Reply by jeet-proj** [reply]
These are the results of config and doctor 
```console
$ brew config
HOMEBREW_VERSION: 3.1.8
ORIGIN: https://github.com/Homebrew/brew
HEAD: e0bfcdaf258dadaaf8c102d4cb51f23b6fdb2c4c
Last commit: 18 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 3c56c184ae6e2d71268c038f949532eac98cd82c
Core tap last commit: 12 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
CPU: quad-core 64-bit ivybridge
Clang: 12.0.0 build 1200
Git: 2.24.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 10.15.7-x86_64
CLT: 12.0.0.32.29
Xcode: 12.4
```

------
```console
$ brew doctor
Your system is ready to brew.
```
